src/runtime/cpu/kernels/kernels.cpp

```diff
@@ -46,6 +46,17 @@ void deepworks::CPUReLUInputGrad(ConstVector input,
     grad_input = (input.array() > 0.f).select(grad_output, 0.f);
 }
 
+void deepworks::CPULeakyReLUForward(ConstVector X, Vector result, float alpha) {
+    result = (X.array() < 0.f).select(X.array() * alpha, X);
+}
+
+void deepworks::CPULeakyReLUInputGrad(ConstVector input, ConstVector grad_output,
+                                      Vector grad_input, float alpha) {
+    Eigen::MatrixXf ones = Eigen::MatrixXf::Ones(input.outerSize(), input.innerSize());
+    grad_input = grad_output.array() *
+                 (input.array() < 0.f).select(alpha, ones).array();
```

Comment:
Can we do this without creating a ones matrix?
```cpp
    grad_input = input.array() < 0.f).select(alpha * grad_output, grad_output).array();
```

Reply:
Done.